pkg/downloader/types.go

```diff
@@ -0,0 +1,16 @@
+package downloader
+
+import (
+	"context"
+
+	"github.com/filecoin-project/bacalhau/pkg/model"
+)
+
+type Downloader interface {
+	// FetchResult fetches result contained in PublishedShardDownloadContext
+	FetchResult(ctx context.Context, shardCidContext model.PublishedShardDownloadContext) error
```

Comment:
Is `PublishedShardDownloadContext` definitely the right type to be passing to a Downloader? The current implementations only ever seem to access the `Result` and the `CIDDownloadDir`. Would it be better to only pass those things?

One motivation for this is that we currently also have `pkg/storage/url/urldownload/storage.go` which downloads things from URLs for _input_. That currently takes just a StorageSpec. I can see a world where we (in the future) unify the storage providers and the downloader providers, so that you only have to implement one of them to make it work on both server and client. So having a similar interface would make that easier.